tests/test_utils_strict_dataclass.py

```diff
@@ -62,6 +80,42 @@ def test_default_values():
     assert config.hidden_size == 1024
 
 
+def test_forward_ref_validation():
```

Comment:
In this test we are checking much more than just the "forward ref"-logic. Would be good to make it a minimal unit test